examples/infra/mod.rs

```diff
@@ -54,8 +58,281 @@ use libp2p::{
+
+    // Generate network's public keys
+    config.config.known_nodes = (0..config.config.total_nodes.get())
+        .map(|node_id| {
+            TYPES::SignatureKey::generated_from_seed_indexed(
+                config.seed,
+                node_id.try_into().unwrap(),
+            )
+            .0
+        })
+        .collect();
+
+    config
+}
+
+/// Runs the orchestrator
+pub async fn run_orchestrator<
+    TYPES: NodeType,
+    MEMBERSHIP: Membership<TYPES>,
+    NETWORK: CommunicationChannel<
+        TYPES,
+        ValidatingProposal<TYPES, ValidatingLeaf<TYPES>>,
+        QuorumVote<TYPES, ValidatingLeaf<TYPES>>,
+        MEMBERSHIP,
+    >,
+    NODE: NodeImplementation<
+        TYPES,
+        Leaf = ValidatingLeaf<TYPES>,
+        Proposal = ValidatingProposal<TYPES, ValidatingLeaf<TYPES>>,
+        Membership = MEMBERSHIP,
+        Networking = NETWORK,
+        Storage = MemoryStorage<TYPES, ValidatingLeaf<TYPES>>,
+    >,
+>(
+    OrchestratorArgs {
+        host,
+        port,
+        config_file,
+    }: OrchestratorArgs,
+) {
+    error!("Starting orchestrator",);
+    let run_config = load_config_from_file::<TYPES>(config_file);
+    let _result = hotshot_orchestrator::run_orchestrator::<
+        TYPES::SignatureKey,
+        TYPES::ElectionConfigType,
+    >(run_config, host, port)
+    .await;
+}
+
+// VALIDATOR
+
+#[derive(Parser, Debug, Clone)]
+#[command(
+    name = "Multi-machine consensus",
+    about = "Simulates consensus among multiple machines"
+)]
+/// Arguments passed to the validator
+pub struct ValidatorArgs {
+    /// The address the orchestrator runs on
+    host: IpAddr,
+    /// The port the orchestrator runs on
+    port: u16,
+}
+
+/// Defines the behavior of a "run" of the network with a given configuration
+#[async_trait]
+pub trait Run<
+    TYPES: NodeType,
+    MEMBERSHIP: Membership<TYPES>,
+    NETWORK: CommunicationChannel<
+        TYPES,
+        ValidatingProposal<TYPES, ValidatingLeaf<TYPES>>,
+        QuorumVote<TYPES, ValidatingLeaf<TYPES>>,
+        MEMBERSHIP,
+    >,
+    NODE: NodeImplementation<
+        TYPES,
+        Leaf = ValidatingLeaf<TYPES>,
+        Proposal = ValidatingProposal<TYPES, ValidatingLeaf<TYPES>>,
+        Membership = MEMBERSHIP,
+        Networking = NETWORK,
+        Storage = MemoryStorage<TYPES, ValidatingLeaf<TYPES>>,
+    >,
+> where
+    <TYPES as NodeType>::StateType: TestableState,
+    <TYPES as NodeType>::BlockType: TestableBlock,
+    ValidatingLeaf<TYPES>: TestableLeaf,
+    HotShot<TYPES::ConsensusType, TYPES, NODE>: ViewRunner<TYPES, NODE>,
+    Self: Sync,
+{
+    /// Initializes networking, returns self
+    async fn initialize_networking(
+        config: NetworkConfig<TYPES::SignatureKey, TYPES::ElectionConfigType>,
+    ) -> Self;
+
+    /// Initializes the genesis state and HotShot instance; does not start HotShot consensus
+    async fn initialize_state_and_hotshot(&self) -> (TYPES::StateType, HotShotHandle<TYPES, NODE>) {
```

Comment:
nit: might make sense to have a `# Panics` section in here. Looks like this only panics if we can't generate a genesis block, fail to init, or fail to get the anchored view?